Update documentation to make the possible settings for state_characteristic more searchable (#26596)
home-assistant/home-assistant.github.io: source/_integrations/statistics.markdown

## Characteristics

> The following statistical characteristics (`state_characteristic`) are available. Pay close attention to the right configuration of `sampling_size` and/or `max_age`, as most characteristics are directly influenced by these settings.

Author: Not a fan of how this flows.  Suggestions welcome.

> The following statistical characteristics (`state_characteristic`) are available. Pay close attention to the right configuration of `sampling_size` and/or `max_age`, as most characteristics are directly influenced by these settings.

Reviewer: ```suggestion
The following statistical characteristics (`state_characteristic`) are available. Pay close attention to the correct configuration of `sampling_size` and/or `max_age`, as most characteristics are directly influenced by these settings.
```
While maybe not perfect, it seems reasonable to me.

Reviewer 2: Agree with this.

Reviewer 2: ```suggestion
The following options for the configuration parameter `state_characteristic` are available as statistical characteristics. Pay close attention to the right configuration of `sampling_size` and/or `max_age`, as most characteristics are directly influenced by these settings.
```

Author: I used a mix of both the suggestions (changing 'right' -> 'correct'), but using the same sentence structure as above.

### Numeric Source Sensor

> The following are supported for `sensor` source sensors `state_characteristic`:

Reviewer: I think this one flows worse (same for the binary further below).   Can the word "source" be eliminated?  I know the following isn't perfect, but maybe something like:
```
The following `state_characteristic`s are supported by the `sensor` type.
```

Reviewer 2: This documentation document talks about the "source sensor" or "source entity" consistently, to establish a clear difference between the created statistics sensor, and the underlying source sensor it is derived from.

```suggestion
The following options for `state_characteristic` are supported for source entities of the `sensor` domain:
```

Author: I'm getting mixed signals here.  How should I proceed?

### Numeric Source Sensor

> The following are supported for `sensor` source sensors for `state_characteristic`:

Reviewer 2: Could you please get rid of the double "for" ?

### Binary Source Sensor

> The following are supported for `binary_sensor` source sensors for `state_characteristic:

Reviewer 2: Same as above

  default: Stats
  type: string
state_characteristic:

>   description: The characteristic that should be used as the state of the statistics sensor (see tables above for each sensor type).

Reviewer 2: "tables" agreed, but the rest can be removed I believe
